src/components/RightPane/CoursePane/SearchForm/SectionCodeSearchBar.tsx

```diff
@@ -3,6 +3,11 @@ import React, { ChangeEvent, PureComponent } from 'react';
 
 import RightPaneStore from '../../RightPaneStore';
 
+const urlParamValue = new URLSearchParams(window.location.search).get("courseCode");
```

Comment:
@bevm0 Do you know if this line being outside any function or class creates race conditions with orders of module loading or is it guaranteed to be loaded after `RightPaneStore` since it imports it, or is it at the same time because it's in the same bundle?

Reply:
Hmm, honestly if it's even a consideration, I'd wager that the strategy could be improved.

I actually don't know if the solution could be viable at all because the variable lives completely outside of the component life-cycle, which is terrible for a client-side rendered app that only runs the render method whenever components update; it never runs the rest of the function, so I speculate this only works onMount and never again afterwards.

I haven't tried it, but I think an idiomatic solution would be to keep track of the `urlParamValue` as a state variable, then use `useEffect` (or whatever accomplishes the same thing) to listen to `window.location.search` and then update that state variable to the latest value.